### DISCUSSION TITLE
network.WLAN - how to read docs?

### DISCUSSION THREAD

**Discussion by shorted-neuron** [micropython/micropython#11432] [Using MicroPython] [answered]
I'm getting started with micropython on a Raspi rp2 Pico W.  I'm a noob and having trouble understanding the documentation, and have been unsuccessful in finding "how to read the documentation" documentation.

Specifically, this document is at the heart of my issues: [https://docs.micropython.org/en/latest/library/network.WLAN.html](https://docs.micropython.org/en/latest/library/network.WLAN.html). I am having trouble getting my wifi going, following [https://datasheets.raspberrypi.com/picow/connecting-to-the-internet-with-pico-w.pdf](https://datasheets.raspberrypi.com/picow/connecting-to-the-internet-with-pico-w.pdf) esp section 3.6 "Connecting to a wireless network"

**_issue 1 -_**   Where do i find how to translate the constants there to their values?  network.WLAN.status() in that doc shows constants.  I am getting several values, 2, -2, -3, but never "3" which in examples seems to be the value indicating a successful connection.

Similarly, i thought to try the WLAN.connect() method.  In many examples, you invoke it with 
```WLAN.connect(ssid, psk)```

I thought as part of troubleshooting, that i would specify the access point.  The documentation says:
```WLAN.connect(ssid=None, key=None, *, bssid=None)```

**_Issue 2 - what is the third argument,_** ```*``` ?  I do not know how to interpret that, and it seems that I'm "supposed to know" what that means.  I cannot invoke the method without it... coming from C, i thought it might be a pointer, string, etc.  nothing i try passes the interpreter.

_**Issue 3 -**_ most important, i'd like to know how I might have found these answers myself if i were more experienced!

thanks a pile!

**Comment by peterhinch** [comment, marked as answer]
To connect to WiFi use something like this (changing network name and password):
```py
def do_connect():
    import network
    sta_if = network.WLAN(network.STA_IF)
    ap = network.WLAN(network.AP_IF) # create access-point interface
    ap.active(False)         # deactivate the interface
    if not sta_if.isconnected():
        print('connecting to network...')
        sta_if.active(True)
        sta_if.connect("NETWORK_NAME", "PASSWORD")
        while not sta_if.isconnected():
            pass
    print('network config:', sta_if.ifconfig())
    a = sta_if.config('mac')
    print('MAC {:02x}:{:02x}:{:02x}:{:02x}:{:02x}'.format(a[0],a[1],a[2],a[3],a[4]))
```
In general you don't need to worry about the numeric values of the constants.

The `*` in the calling convention is part of the Python language. The compiler takes it to mean that subsequent parameters are named rather than positional. The function:
```py
def foo(x, y, *, z):
    print(x, y, z)
```
must be called with `z` specified by name. Either of the following will work:
```py
foo(1, 2, z=3)
foo(1, y=2, z=3)
```
but not
```py
foo(1, 2, 3)
```
Language details are best learnt by following an online course on your PC.

**Reply by shorted-neuron** [reply]
Thank you very much sir, that helped.  My code was similar enough to the documentation examples and your own answer that I started to suspect a bad pico unit.  After much trial and error, and another pico W displaying same behavior, I started using a different access point.  Then i suddenly was able to connect very reliably.  I have not figured out what the difference is between the two, as both are openwrt with very similar configurations, but if i ever do, i will post here.

So... for anyone that happens to read this, if the default examples are giving you trouble with WiFi, you might consider trying a different access point (entirely different Wifi network).

Thanks again!
